### ISSUE TITLE
[dev 5552] Tab below the stack receives focus for a fraction of a second after closing the last tab in stack

### ISSUE THREAD

**Issue by SXZ1** [piroor/treestyletab#1510]
This bug might be a fallout from this commit: https://github.com/piroor/treestyletab/commit/bafe216d5a5d875fadea4a1febf00058e4d3b3b6
Create new profile, install latest TST dev https://piro.sakura.ne.jp/xul/xpi/nightly/treestyletab-we.xpi, close-launch Nightly, go to github.com in the active tab, ctrl-click Features, ctrl-click Business
Open new tab, go to https://forums.geforce.com/, press Ctrl-2, press Ctrl-3, press Ctrl-W

ER: Geforce tab gets focus, that's how Firefox behaves without TST installed
AR: Geforce tab gets focus for a fraction of a second, then Business tab gets focus

Scheme by @gnite:

    A
     B
     C (active)
    D

> After you close C, D becomes active for a split second before switching to B as expected. Sometimes it's annoying when D was previously unloaded and has some autoplaying video as example.

**Comment by piroor** [comment]
This is not a bug, but a designed (intentional) behavior. In other words, the tab D should not be focused in this case because the closed tab C is a child tab and still there is more other child tab B. On such case TST tries to keep focus in same tree, so B becomes new active tab. After B is closed, A will be focused by same reason.

On the other hand, Firefox always tries to move focus to next tab if the active tab is closed. Because WebExtensions addon cannot cancel the behavior, TST just wait for Firefox to complete its "closing current tab" operation, and moves focus again.

**Comment by gnite** [comment]
Of course, switching from C to B and then to A is exactly as it should be, but the jump to D in between can sometimes be weird. As to it being impossible to change this behavior, the one thing I can say is that when using the other Tree Tabs add-on for a while this did not happen, so there must be some way around that.

**Comment by confluence** [comment]
What is the correct TST setting for preserving the default Firefox behaviour (i.e. always changing focus to the next tab) when a tab is closed from outside the tab bar (or under any circumstances, since I never actually click on the close buttons to close tabs)?

I have changed the settings several times, and thought I had fixed it, only to have the issue return -- possibly I'm just not correctly reproducing some TST-specific situation.

If there isn't a setting for this, I would like to request one. This always just worked as expected in pre-57 TST, and the new behaviour is driving me crazy.
